### ISSUE TITLE
get_column_type refactor

### ISSUE THREAD

**Issue by pmbrull** [open-metadata/OpenMetadata#3582]
**Affected module**
Ingestion

**Describe the bug**
It is quite hard to follow and debug what is going on in get_column_type at column_type_parser. Let's break the function down into smaller functions with meaningful names, comments, and tests. We can refactor the protected functions of the module later on, but this one gets directly called by the workflow so we should first focus on this.

It seems to me that there are 3 main steps involved. Convert each of them in a function and document the order of "checks" that we are doing.

Thanks!

**Comment by ayush-shah** [comment]
Hey @pmbrull, Thanks for creating this issue.
Will take it up in 0.9.2 release

**Comment by pmbrull** [comment]
Great, thanks @ayush-shah, no rush at all :) Let's just try to list down central code pieces that we could improve. Thanks 🙏

**Comment by harshach** [comment]
@pmbrull @ayush-shah do we need to do anything here?

**Comment by harshach** [comment]
@pmbrull @ayush-shah what is remaining here

**Comment by ayush-shah** [comment]
Refactoring the code, breaking it down in functions,
Writing tests and adding upto 100% test coverage around here.
Will be part of 0.13

**Comment by harshach** [comment]
@ayush-shah is this targetted for 0.13?

**Comment by ayush-shah** [comment]
yes
